x-pack/plugins/uptime/server/lib/requests/get_journey_failed_steps.ts

```diff
@@ -53,11 +53,17 @@ export const getJourneyFailedSteps: UMElasticsearchQueryFn<GetJourneyStepsParams
 
   const { body: result } = await uptimeEsClient.search({ body: params });
 
-  return ((result.hits.hits as Array<SearchHit<Ping>>).map((h) => {
-    const source = h._source as Ping & { '@timestamp': string };
+  return result.hits.hits.map(({ _id, _source }) => {
+    const decoded = JourneyStepType.decode(Object.assign({ _id }, _source));
```

Comment:
This is a pattern I've seen repeated several times just in the code of this PR. Does anyone know if there's a utility in Kibana for handling this type of error? If not, we should make a common util for it? Runtime type checking ES responses will cut down significantly on our usage of `any`, and the types it creates are reusable for the client.

Reply:
i am not sure why we need this validaiton on server side?